browser/brave_news/brave_news_tab_helper.cc

```diff
@@ -83,36 +84,30 @@ const std::vector<GURL> BraveNewsTabHelper::GetAvailableFeedUrls() {
 }
 
 bool BraveNewsTabHelper::IsSubscribed(const GURL& feed_url) {
-  auto* publisher =
-      controller_->publisher_controller()->GetPublisherForFeed(feed_url);
-  if (!publisher) {
-    return false;
+  if (page_feed_ && page_feed_->feed_source == feed_url) {
+    return brave_news::IsSubscribed(page_feed_);
   }
 
-  // When a direct feed exists, it is always subscribed (there is no way to
-  // have an unsubscribed direct feed).
-  if (publisher->type == brave_news::mojom::PublisherType::DIRECT_SOURCE) {
-    return true;
+  auto it = std::ranges::find(rss_page_feeds_, feed_url,
+                              [](const auto& feed) { return feed.feed_url; });
+  if (it == rss_page_feeds_.end()) {
+    return false;
   }
 
-  // Otherwise, it's a combined feed, so just return whether the user has
-  // enabled it.
-  return publisher->user_enabled_status ==
-         brave_news::mojom::UserEnabled::ENABLED;
+  return it->subscribed;
```

Comment:
Is this when users subscribe a topic not individual publishers? Actually I don't seem to understand what direct feed is.

Reply:
So we do a bunch of preprocessing in our backend and generate a `CombinedFeed` which aggregates some of the most popular publishers into a big feed. Direct feeds are where a user subscribes directly to an RSS feed on the page.

Reply:
Thanks for the explanation!